bin/add_transform_test.py

```diff
@@ -0,0 +1,148 @@
+                properties["EndpointConfiguration"] = {"Types": ["REGIONAL"]}
+            if "Parameters" not in properties:
+                properties["Parameters"] = {"endpointConfigurationTypes": "REGIONAL"}
+
+    return template
+
+
+def generate_transform_test_output_files(input_file_path: str, file_basename: str) -> None:
+    output_file_option = file_basename + ".json"
+
+    # run sam-translate.py and get the temporary output file
+    with tempfile.NamedTemporaryFile() as temp_output_file:
+        subprocess.run(
+            [
+                sys.executable,
+                SCRIPT_DIR + "/sam-translate.py",
+                "--template-file",
+                input_file_path,
+                "--output-template",
+                temp_output_file.name,
+            ],
+            check=True,
+        )
+
+        # copy the output files into correct directories
+        transform_test_output_path = TRANSFORM_TEST_DIR + "/output/" + output_file_option
+        shutil.copyfile(temp_output_file.name, transform_test_output_path)
+
+        regional_transform_test_output_paths = [
+            TRANSFORM_TEST_DIR + path + output_file_option
+            for path in [
+                "/output/aws-cn/",
+                "/output/aws-us-gov/",
+            ]
+        ]
+
+        if not CLI_OPTIONS.get("--disable-api-configuration"):
+            template = read_json_file(temp_output_file.name)
+            template = add_regional_endpoint_configuration_if_needed(template)
+            write_json_file(template, temp_output_file.name)
+
+        for output_path in regional_transform_test_output_paths:
+            shutil.copyfile(temp_output_file.name, output_path)
+            print(f"Transform Test output files generated {output_path}")
+
+
+def map_nested(obj: Any, fn) -> Any:
+    if isinstance(obj, dict):
+        return {k: map_nested(v, fn) for k, v in obj.items()}
+    if isinstance(obj, list):
+        return [map_nested(v, fn) for v in obj]
+    return fn(obj)
+
+
+def replace_arn(s: Any) -> Any:
+    if not isinstance(s, str):
+        return s
+
+    pattern = "arn:aws:"
+    replaced_pattern = "arn:${AWS::Partition}"
+    if pattern in s:
+        # pattern is substring of s, use Fn::Sub to replace part of s
+        s = s.replace(pattern, replaced_pattern)
+    if re.search(r"\${.+}", s):
+        return {"Fn::Sub": s}
+    return s
+
+
+def replace_arn_partitions(input_file_path: str) -> None:
+    with open(input_file_path, "r") as f:
+        sam_template = yaml_parse(f)
+
+    replaced_template = map_nested(sam_template, lambda v: replace_arn(v))
+
+    with open(input_file_path, "w") as f:
+        yaml.dump(replaced_template, f, default_flow_style=False)
+
+
+def main() -> None:
+    input_file_path = get_input_file_path()
+    file_basename = Path(input_file_path).stem
+
+    transform_test_input_path = TRANSFORM_TEST_DIR + "/input/" + file_basename + ".yaml"
```

Comment:
how about windows dev who wants to create them

Reply:
Good catch. Updated to use `os.path.join`